File: backend/internal/model/project_plan.go
package model

import "time"

type ProjectModule struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"size:100;not null" json:"name"`
	Icon      string    `gorm:"size:10" json:"icon"`
	SortOrder int       `gorm:"default:0" json:"sort_order"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

type ProjectItem struct {
	ID           uint      `gorm:"primaryKey" json:"id"`
	ModuleID     uint      `gorm:"not null;index" json:"module_id"`
	Name         string    `gorm:"size:200;not null" json:"name"`
	Status       string    `gorm:"size:20;default:'not_started'" json:"status"`
	Priority     string    `gorm:"size:20;default:'medium'" json:"priority"`
	Owner        string    `gorm:"size:50" json:"owner"`
	PlannedHours *float64  `json:"planned_hours"`
	ActualHours  *float64  `json:"actual_hours"`
	Description  string    `gorm:"type:text" json:"description"`
	SortOrder    int       `gorm:"default:0" json:"sort_order"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

func (ProjectModule) TableName() string {
	return "project_modules"
}

func (ProjectItem) TableName() string {
	return "project_items"
}

File: frontend/src/api/project-plan.ts
import http from './http'

const request = {
  get: <T = any>(url: string, config?: any) => http.get<T>(url, config),
  post: <T = any>(url: string, data?: any, config?: any) => http.post<T>(url, data, config),
  put: <T = any>(url: string, data?: any, config?: any) => http.put<T>(url, data, config),
  delete: <T = any>(url: string, config?: any) => http.delete<T>(url, config)
}

export interface ProjectModule {
  id: number
  name: string
  icon: string
  sort_order: number
  created_at: string
  updated_at: string
}

export interface ProjectItem {
  id: number
  module_id: number
  name: string
  status: 'not_started' | 'in_progress' | 'completed' | 'blocked'
  priority: 'low' | 'medium' | 'high' | 'urgent'
  owner: string
  estimated_time: string
  notes: string
  sort_order: number
  created_at: string
  updated_at: string
}

export interface ModuleWithItems extends ProjectModule {
  items: ProjectItem[]
}

export const getModulesWithItems = () => {
  return request.get<ModuleWithItems[]>('/project-plan/modules')
}

export const createModule = (data: Partial<ProjectModule>) => {
  return request.post<ProjectModule>('/project-plan/modules', data)
}

export const updateModule = (id: number, data: Partial<ProjectModule>) => {
  return request.put<ProjectModule>(`/project-plan/modules/${id}`, data)
}

export const deleteModule = (id: number) => {
  return request.delete(`/project-plan/modules/${id}`)
}

export const createItem = (data: Partial<ProjectItem>) => {
  return request.post<ProjectItem>('/project-plan/items', data)
}

export const updateItem = (id: number, data: Partial<ProjectItem>) => {
  return request.put<ProjectItem>(`/project-plan/items/${id}`, data)
}

export const deleteItem = (id: number) => {
  return request.delete(`/project-plan/items/${id}`)
}

export const batchUpdateItems = (items: ProjectItem[]) => {
  return request.post('/project-plan/items/batch', items)
}
